pkg/blueprint/disk_customizations.go

```diff
@@ -18,6 +18,27 @@ type DiskCustomization struct {
 	Partitions []PartitionCustomization `json:"partitions,omitempty" toml:"partitions,omitempty"`
 }
 
+func (dc *DiskCustomization) UnmarshalJSON(data []byte) error {
```

Comment:
meh, this lacks the toml unmarshaler :(

Reply:
Also needs some error tests

Reply:
We probably can close this in favor of https://github.com/osbuild/images/pull/1057 which archives this with less code (but  some boilerplate).